File: C/stack/stack.c
//skdhg

#include <stdio.h>

int MAXSIZE = 8;
int stack[8];
int top = -1;

/* Check if the stack is full */
int isFull() {
   if (top == MAXSIZE-1)
      return 1;
   else
      return 0;
}

/* Check if the stack is empty */
int isEmpty() {
   if (top == -1)
      return 1;
   else
      return 0;
}

/* Add elements into stack */
void push(int data) {
   if (!isFull()) {
      top = top + 1;
      stack[top] = data;
   } else {
      printf("Could not insert data, Stack is full.\n");
   }
}

/* Remove element from stack */
int pop() {
   int data;

   if (!isEmpty()) {
      data = stack[top];
      top = top - 1;
      return data;
   } else {
      printf("Could not retrieve data, Stack is empty.\n");
   }
}

/* Returns element at top of stack */
int peek() {
   return stack[top];
}

/* Display the stack */
void display() {
   int i;

   if (!isEmpty()) {
      for (i = top; i >= 0; i--)
         printf("%d ", stack[i]);
      printf("\n");
   } else {
      printf("Stack is empty.\n");
   }
}

/* Search element in stack */
void search(int element) {
   printf("-------------------Search Operation------------------- \n");

   for (int i = 0; i < top; i++) {
      if (stack[i] == element) {
         printf("Element found at index %d \n", i);
         return;
      }
   }

   printf("Element not found \n");
}

int main() {
   // push items on to the stack
   push(3);
   push(5);
   push(9);
   push(1);
   push(12);
   push(15);

   printf("Element at top of the stack: %d\n", peek());
   printf("Elements: \n");

   // print stack data
   display();

   // remove element from stack
   printf("\nElement popped: %d\n", pop());
   printf("Element at top of the stack: %d\n", peek());
   printf("Elements: \n");

   // print stack data
   display();

   // search element in stack
   search(9);

   return 0;
}

File: Java/stack/Stack.java
public class Stack {
    static int MAXLEN = 10;
    static int top = -1;
    static int[] stack = new int[MAXLEN];

    public static void main(String[] args) {
        // push items on to the stack
        push(3);
        push(5);
        push(9);
        push(1);
        push(12);
        push(15);

        System.out.printf("Element at top of the stack: %d\n", peek());
        System.out.println("Elements:");

        // print stack data
        display();

        // remove element from stack
        System.out.printf("\nElement popped: %d\n", pop());
        System.out.printf("Element at top of the stack: %d\n", peek());
        System.out.println("Elements:");

        // print stack data
        display();
    }

    public static void push(int item) {
        if (isFull()) {
            System.out.println("Cannot push, stack is full!");
            return;
        }

        stack[++top] = item;
    }

    public static int pop() {
        if (isEmpty()) {
            System.out.println("Cannot pop, stack is empty!");
            return 0;
        }

        return stack[top--];
    }

    public static int peek() {
        if (isEmpty()) {
            System.out.println("Cannot peek, stack is empty");
            return 0;
        }

        return stack[top];
    }

    public static void display() {
        System.out.print("[");
        for(int i = 0; i <= top; i++) {
            System.out.printf("%d", stack[i]);
            if (i < top) System.out.print(", ");
        }
        System.out.println("]");
    }

    public static boolean isFull() {
        return top == MAXLEN;
    }

    public static boolean isEmpty() {
        return top == -1;
    }
}
